src/Orchard.Web/Modules/Orchard.Projections/Services/ProjectionManager.cs

```diff
@@ -112,9 +116,15 @@ public class ProjectionManager : IProjectionManager{
             }
 
             // aggregate the result for each group query
+            var checkPermissions = queryRecord.FilterGroups.Any(x => x.Filters.Any(y => y.CheckPermissions));
 
-            return GetContentQueries(queryRecord, Enumerable.Empty<SortCriterionRecord>())
-                .Sum(contentQuery => contentQuery.Count());
+            var count = GetContentQueries(queryRecord, Enumerable.Empty<SortCriterionRecord>())
+                .Sum(contentQuery => checkPermissions ? 
+                                contentQuery.List().FilterContentItems(_services.WorkContext.CurrentUser).Count() : 

```

Comment:
So, in my tests, I set the page size to 5 and had 11 content items.  5 of which were private with different sets of permissions.  When I was not logged in, it successfully displayed 2 pages (6 content items for anonymous access).  When I was logged in, it displayed all 11 in 3 pages.

So yes, my concern is that it is checking all items, not just up to the current page... because the page is dynamic now, depending on the number of items the user is authorized to see.

If I have the following:
* contentItem1 public
* contentItem2 public
* contentItem3 private
* contentItem4 public
* contentItem5 private
* contentItem6 public
* contentItem7 private
* contentItem8 public
* contentItem9 private
* contentItem10 public
* contentItem11 private

Logged in, I would see all of them (with the proper role, obviously) split across 3 pages of 5, 5, 1 items

Logged out, I would see:
Page 1:
* contentItem1 public
* contentItem2 public
* contentItem4 public
* contentItem6 public
* contentItem8 public

Page 2:
* contentItem10 public